### PR TITLE
chore: increase some coverage with the plots

### PR DESCRIPTION
**Pull request by thatlittleboy**
# Changes

* raise `DimensionError` instead of `AssertionError` when features shape don't match the shap values' shape. + test for this.
  * beeswarm
  * violin
* If a non-Explanation object is passed to `shap.plots.beeswarm`, raise `TypeError` instead of `ValueError`
* Add a basic test for `shap.plots.beeswarm`.

### CODE REVIEW

File: shap/plots/_beeswarm.py [hunk 1]
```diff
@@ -43,7 +44,11 @@ def beeswarm(shap_values, max_display=10, order=Explanation.abs.mean(0),
     """
 
     if not isinstance(shap_values, Explanation):
-        raise ValueError("the beeswarm plot requires Explanation object as the `shap_values` argument")
+        emsg = (
+            "The beeswarm plot requires an `Explanation` object as the "
+            "`shap_values` argument."
+        )
+        raise TypeError(emsg)
```
Comment (connortann) [inline shap/plots/_beeswarm.py:51]: Nitpick: for simplicity, could avoid creating the variable `emsg` and just do:
```python
raise TypeError(
   "The beeswarm plot ..."
)
```

**Review by connortann — APPROVED** [review]
Overall LGTM

File: shap/plots/_beeswarm.py (hunk 1 above)
Comment (dsgibbons) [reply shap/plots/_beeswarm.py:51]: > Nitpick: for simplicity, could avoid creating the variable emsg and just do:

Ruff TRY003 doesn't allow long messages inside exceptions.

Comment (connortann) [reply shap/plots/_beeswarm.py:51]: I don't think that particular errors group is enabled in our linter configuration. From a quick look, the `tryceratops` rules seem a little strict. I gather the aim in this instance is to encourage the use of custom Exception class.

Are you running the `TRY` set of checks on Ruff locally, and is it a set of rules that you'd advocate for?

Comment (thatlittleboy) [reply shap/plots/_beeswarm.py:51]: The reason why I separate out the error messages is due to `flake8-errmsg`. You can see the reasoning here: https://beta.ruff.rs/docs/rules/raw-string-in-exception/.

**Pull request merged**